### PR TITLE
Update reviewing and merging documentation

### PR DESCRIPTION
**Pull request by brentswisher**
The list of open pull requests for Gutenberg keeps growing. This is an attempt to update the developer documentation to encourage more people to participate in the reviewing and merging process, as discussed in the recent developer meeting - https://wordpress.slack.com/archives/C02QB2JS7/p1564580809002700

### CODE REVIEW

File: docs/contributors/repository-management.md [hunk 1]
```diff
@@ -116,6 +116,8 @@ Every pull request goes through a manual code review, in addition to automated t
 
 *As a contributor*, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.
 
+Code reviews are encouraged by everyone who is willing to attempt one. If you review a pull request and are confident in the changes, approve it. If you don't feel totally confident it is ready for merging, add your review with a comment that says it should have another set of eyes on it before final approval. This can help filter out obvious bugs and simplify reviews for core members. It also is a great learning experience as you can follow the later reviews to see how to improve your own future reviews.
+
```
Comment (tellthemachines) [inline docs/contributors/repository-management.md]: Might be good to add that it's also fine to just comment with questions when you're unsure of how something should work or if the reason for the change is unclear. Or review only the parts of the code you are comfortable with, and comment that changes x and y haven't been looked at yet. 
Great to emphasise code review as a learning experience 🙂

File: docs/contributors/repository-management.md [hunk 2]
```diff
@@ -138,7 +140,10 @@ A pull request can generally be merged once it is:
 
 The final pull request merge decision is made by the **@wordpress/gutenberg-core** team.
 
-Please make sure to assign your merged pull request to its release milestone. Doing so creates the historical legacy of what code landed when, and makes it possible for all project contributors (even non-technical ones) to access this information.
+All members of the Wordpress organization on Github have the ability to review and merge pull requests. If you have reviewed a PR and are confident in the code, approve the pull request and comment pinging **@wordpress/gutenberg-core** or a specific core member who has been involved in the PR. Once they confirm there are no objections, you are free to merge the PR into master.
```
Comment (gziolo) [inline docs/contributors/repository-management.md]: Typos:
`Wordpress` -> `WordPress`
`Github` -> `GitHub`

**Comment by jasmussen** [comment]
With the changes suggested already, this seems good to me 👍 👍

File: docs/contributors/repository-management.md (hunk 2 above)
Comment (brentswisher) [reply docs/contributors/repository-management.md]: Thanks! Updated

File: docs/contributors/repository-management.md (hunk 1 above)
Comment (brentswisher) [reply docs/contributors/repository-management.md]: Thanks for the feedback! I took a stab at what you were suggesting, let me know what you think.

Comment (tellthemachines) [reply docs/contributors/repository-management.md]: Looks great, thanks for this!

**Review by tellthemachines — APPROVED** [review]

**Review by mapk — APPROVED** [review]
:shipit:

**Pull request merged**